### PR TITLE
[Merged by Bors] - refactor(Topology/Order) : Fix inconsistencies between Lower/Upper and Scott topologies

### PR DESCRIPTION
**Pull request by mans0954**
- Make type explicit in def of Scott-Hausdorff and Scott topologies
- Move `With{Lower|Upper}Homeomorph` into the correct namespace

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Order/LowerUpperTopology.lean [hunk 1]
```diff
@@ -389,6 +384,12 @@ end PartialOrder
 
 end IsUpper
 
+/-- If `α` is equipped with the upper topology, then it is homeomorphic to `WithUpper α`.
+-/
+def WithUpper.withUpperHomeomorph [Preorder α] [TopologicalSpace α] [IsUpper α] :
+    WithUpper α ≃ₜ α :=
+  WithUpper.ofUpper.toHomeomorphOfInducing ⟨by erw [IsUpper.topology_eq α, induced_id]; rfl⟩
```
Comment (YaelDillies) [inline Mathlib/Topology/Order/LowerUpperTopology.lean:391]: I think 
```suggestion
def IsUpper.withUpperHomeomorph [Preorder α] [TopologicalSpace α] [IsUpper α] :
    WithUpper α ≃ₜ α :=
  WithUpper.ofUpper.toHomeomorphOfInducing ⟨by erw [IsUpper.topology_eq α, induced_id]; rfl⟩
```
is better

Comment (mans0954) [reply Mathlib/Topology/Order/LowerUpperTopology.lean:391]: Presumably it should then be `IsUpper.withScottHomeomorph` to match?

Comment (mans0954) [reply Mathlib/Topology/Order/LowerUpperTopology.lean]: Done.

File: Mathlib/Topology/Order/ScottTopology.lean [hunk 2]
```diff
@@ -325,27 +327,30 @@ instance [Inhabited α] : Inhabited (WithScott α) := ‹Inhabited α›
...
 
 instance : Preorder (WithScott α) := ‹Preorder α›
-instance : TopologicalSpace (WithScott α) := scott
+instance : TopologicalSpace (WithScott α) := scott α
 instance : IsScott (WithScott α) := ⟨rfl⟩
 
 lemma isOpen_iff_isUpperSet_and_scottHausdorff_open' {u : Set α} :
-    IsOpen (WithScott.ofScott ⁻¹' u) ↔ IsUpperSet u ∧ scottHausdorff.IsOpen u := Iff.rfl
+    IsOpen (WithScott.ofScott ⁻¹' u) ↔ IsUpperSet u ∧ (scottHausdorff α).IsOpen u := Iff.rfl
+
```
Comment (YaelDillies) [inline Mathlib/Topology/Order/ScottTopology.lean:336]:
```suggestion
```

File: Mathlib/Topology/Order/LowerUpperTopology.lean [hunk 3]
```diff
@@ -211,7 +211,7 @@ variable {α}
 
 /-- If `α` is equipped with the lower topology, then it is homeomorphic to `WithLower α`.
 -/
-def WithLowerHomeomorph : WithLower α ≃ₜ α :=
+def withLowerHomeomorph : WithLower α ≃ₜ α :=
   WithLower.ofLower.toHomeomorphOfInducing ⟨by erw [topology_eq α, induced_id]; rfl⟩
 #align lower_topology.with_lower_topology_homeomorph Topology.IsLower.WithLowerHomeomorph
```
Comment (YaelDillies) [inline Mathlib/Topology/Order/LowerUpperTopology.lean:216]:
```suggestion
#align lower_topology.with_lower_topology_homeomorph Topology.IsLower.withLowerHomeomorph
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**